driver-core/src/main/com/mongodb/connection/ConnectionId.java

```diff
@@ -50,7 +50,20 @@ public ConnectionId(final ServerId serverId) {
         this(serverId, INCREMENTING_ID.incrementAndGet(), null);
     }
 
-    private ConnectionId(final ServerId serverId, final int localValue, final Integer serverValue) {
+    /**
+     * Construct an instance with the given serverId, localValue, and serverValue.
+     *
+     * <p>
+     *     Useful for testing, but generally prefer {@link #withServerValue(int)}
+     * </p>
+     *
+     * @param serverId the server id
+     * @param localValue the local value
+     * @param serverValue the server value, which may be null
+     * @see #withServerValue(int)
+     * @since 3.11
+     */
+    public ConnectionId(final ServerId serverId, final int localValue, final Integer serverValue) {
```

Comment:
The spec calls for monotonically increasing values per pool, whereas the Java driver implementation is monotonically increasing across the entire VM.  I don't see a reason to change that.

But this makes it possible to test the connection identifiers.  The `TestInternalConnectionFactory` class uses this constructor to assign connection identifiers in the way that's expected of the tests and allows the test runner to assert on connection identifier matches.

Reply:
ACK